### ISSUE TITLE
OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449:

### ISSUE THREAD

**Issue by virt0uz** [kubernetes/kubernetes#96240]
Hi,

Current stack

```
CentOS 7
kubeadm-1.17.11
docker-ce-18.09.2-3
```

We are randomly getting the following error for some of our pods running in our Kubernetes cluster

`Message:      OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/var/lib/kubelet/pods/5c856f54-4a70-49de-9b12-d4a15e12af11/volume-subpaths/lan-truststore/phishing/5\\\" to rootfs \\\"/var/lib/docker/overlay2/cc38f362027461782a32c97ec4b5705c6a46b7d21c8e3a08f0c41637e3bbb954/merged\\\" at \\\"/var/lib/docker/overlay2/cc38f362027461782a32c97ec4b5705c6a46b7d21c8e3a08f0c41637e3bbb954/merged/usr/local/foo/lan_truststore\\\" caused \\\"no such file or directory\\\"\"": unknown`

From our deployment:

```
        volumeMounts:
        - mountPath: /usr/local/foo/lan_truststore
          name: lan-truststore
          subPath: lan_truststore
        - mountPath: /usr/local/foo/lan_password
          name: lan-truststore
          subPath: lan_password
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
      volumes:
...
      - configMap:
          defaultMode: 420
          name: lan-truststore
        name: lan-truststore
```

```
kubectl get cm lan-truststore -o yaml                                                                                                                                                                                                      
apiVersion: v1
binaryData:
  lan_truststore: XXXXXXXXXX
data:
  lan_password: |
    XXXX
```

The issue happens only when the container in the pod restarts. At the first execution it runs fine and mount the correct subpaths.

Any help to troubleshoot this issue would be appreciated,

Thank you in advance,

**Comment by lingsamuel** [comment]
Duplicate #61545?

**Comment by sidd-1** [comment]
Kubernetes version :v1.12.7
OS: Red Hat Enterprise Linux Server release 7.6 (Maipo)
Kernal version: 3.10.0-957.1.3.el7.x86_64

I am also having the same Issue 
```
OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: 
container init caused \"rootfs_linux.go:58: mounting 
\\\"/var/lib/kubelet/pods/1890a27b-4144-11eb-9faa-0a706ee1a05e/volume-subpaths/cacrets/git2consul/1\\\" 
to rootfs \\\"/var/lib/docker/overlay2/b68861c02ac49737d92c7b7b7bcf8fc2dc08c06d96ff39cc4cf1622e5638b36f/merged\\\" at \\\"/var/lib/docker/overlay2/b68861c02ac49737d92c7b7b7bcf8fc2dc08c06d96ff39cc4cf1622e5638b36f/merged/usr/local/openjdk-11/lib/security/cacerts\\\" 
caused \\\"no such file or directory\\\"\"": unknown```
When I restart the pod, it is working. However, it is happening again after some time.
CC: @virt0uz @neolit123

**Comment by nevermore-muyi** [comment]
I met the same problem.
When i mount hostpath of /dev or /lib, the container cannot startup, but when i do not mount hostpath to container, the pod startup successfully.
My kubernetes version is v1.15.6

**Comment by t0rrant** [comment]
I got here while searching for this exact error but directly with docker, not on kubernetes.

Docker Version: 19.03.14

will raise an issue there as well

**Comment by brennerm** [comment]
Encountering the same issue on Kubernetes (EKS) 1.18.9.

`OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/var/lib/kubelet/pods/7718a9bb-5b38-4367-ae2d-8d47c4c64344/volume-subpaths/certificates/0\\\" to rootfs \\\"/var/lib/docker/overlay2/55ace13b7906ec8a06c9078676300aa52e67c776ca90d3b3ada35af1a3728d2f/merged\\\" at \\\"/var/lib/docker/overlay2/55ace13b7906ec8a06c9078676300aa52e67c776ca90d3b3ada35af1a3728d2f/merged/certs/client_keystore.jks\\\" caused \\\"no such file or directory\\\"\"": unknown`

Happens randomly and deleting the containers "solves" the issue.

**Comment by alexandrsemak** [comment]
I will join your party:

`pod/xxxxxxxxx   Error: failed to create containerd task: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/var/data/kubelet/pods/5aa7b825-4192-4240-8563-c7997914f907/volume-subpaths/xxxxxxx/xxxxxx/0\\\" to rootfs \\\"/run/containerd/io.containerd.runtime.v2.task/k8s.io/xxxxx/rootfs\\\" at \\\"/var/data\\\" caused \\\"stat /var/data/kubelet/pods/5aa7b825-4192-4240-8563-c7997914f907/volume-subpaths/xxxxx/xxxxxx/0: no such file or directory\\\"\"": unknown`

**Comment by nevermore-muyi** [comment]
I've solved the problem.
I use privileged mode of my pod and i mount **/dev** directory on the pod. So when removing mount point of **/dev** direcoty, the problem solved.

**Comment by alexandrsemak** [comment]
Yeah I am going left the party. K8s had state races we have bad pod in CronJob which failed and K8s was tried mount secret to terminated pod....

**Comment by shelltea** [comment]
Same problem.

Kubernetes version :v1.16.7
Docker version: 18.09.7
OS: CentOS 7.4
Kernal version: 3.10.0-1160.6.1.el7.x86_64

```
Error: failed to start container "ks-question-service": Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/var/lib/kubelet/pods/fb00e769-b1f3-489c-802b-d611f7037d01/volume-subpaths/skywalking-conf/ks-question-service/5\\\" to rootfs \\\"/data/docker/overlay2/0d5c61704dff3e3c42e3a5605d6b6aec0903d67eda8cc54c388ba4eccef0166f/merged\\\" at \\\"/data/docker/overlay2/0d5c61704dff3e3c42e3a5605d6b6aec0903d67eda8cc54c388ba4eccef0166f/merged/opt/skywalking-agent/config/apm-trace-ignore-plugin.config\\\" caused \\\"no such file or directory\\\"\"": unknown
```

**Comment by fzu-huang** [comment]
Same problem but I can reproduce this case:

1. node with kubelet in  1.11.10-00 
2. deploy a pod with configmap volumes subpath.
3. edit configmap, change file content
4. update kubelet to 1.13.12-00 
5. we can see contianer of this pod is recreating because container hash has change when we upgrade from 1.11 to 1.13
6. new container created but failed to start,  with same log : `mounting XXX to rootfs XXX at XXXXX caused "no such file or directory": unknown`

**Comment by subodh-dharma** [comment]
Facing a similar issue as mentioned in #98719. 

My cluster details:

Kubernetes Version:
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.12", GitCommit:"7cd5e9086de8ae25d6a1514d0c87bac67ca4a481", GitTreeState:"clean", BuildDate:"2020-11-12T09:18:55Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}

Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.12", GitCommit:"7cd5e9086de8ae25d6a1514d0c87bac67ca4a481", GitTreeState:"clean", BuildDate:"2020-11-12T09:11:15Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

Docker: 19.03.13
Kernel: 4.15.0-143-generic
OS: Ubuntu 18

Error details:

```
skipping: failed to "StartContainer" for "api-init" with RunContainerError: "failed to start container \"cef885811c0c74df8a61ca2cee82eab4410e7718a158c3b247e9a034b854e588\": Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused \"process_linux.go:449: container init caused \\\"rootfs_linux.go:58: mounting \\\\\\\"/var/lib/kubelet/pods/7624e759-4afa-4597-9461-0cc6a178e0b2/volume-subpaths/application-settings/api-init/0\\\\\\\" to rootfs \\\\\\\"/var/lib/docker/overlay2/73474fd6375bf6c97cd775b445f8eaad491712e1c82e1b5f715c8432ea51868d/merged\\\\\\\" at \\\\\\\"/var/lib/docker/overlay2/73474fd6375bf6c97cd775b445f8eaad491712e1c82e1b5f715c8432ea51868d/merged/api/cookie/deployment_settings.py\\\\\\\" caused \\\\\\\"no such file or directory\\\\\\\"\\\"\": unknown"
```

VolumeMount Section of the deployment:

```
         volumeMounts:
            - name: application-settings
              mountPath: /api/cookie/deployment_settings.py
              subPath: deployment_settings.py
            - name: application-settings
              mountPath: /api/cookie/authentication_settings.py
              subPath: authentication_settings.py
```

Volumes Section
```
      volumes:
        - name: application-settings
          projected:
            sources:
            - configMap:
                name: api-config
                items:
                - key: deployment_settings.py
                  path: deployment_settings.py
                  mode: 420
            - configMap:
                name: api-config-auth
                optional: true
                items:
                - key: authentication_settings.py
                  path: authentication_settings.py
                  mode: 420

```

**Comment by fzu-huang** [comment]
This problem seems similar to issue: https://github.com/kubernetes/kubernetes/issues/68211  .It has been fixed  in 1.20 , but doesn't backport to 1.19 or earlier versions.   We can find some deleted mountpath on  node which this problem happened:

```
$ cat /proc/1/mountinfo |grep deleted

/var/lib/kubelet/pods/3dc49ca4-b046-11e8-978a-0e51a1d2416a/volume-subpaths/extra-cfg/test/0                                             /dev/xvda9[/var/lib/kubelet/pods/3dc49ca4-b046-11e8-978a-0e51a1d2416a/volumes/kubernetes.io~configmap/extra-cfg/..2018_09_04_13_27_17.387923275/extra.ini//deleted

$ umount /dev/xvda9[/var/lib/kubelet/pods/3dc49ca4-b046-11e8-978a-0e51a1d2416a/volumes/kubernetes.io~configmap/extra-cfg/..2018_09_04_13_27_17.387923275/extra.ini//deleted
```

then subpath will be mount  again and pod will succeefully ran.

**Comment by W1zzardTPU** [comment]
I'm still hitting this on 1.30.7, 6.8.0-49, bare metal,
